src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,106 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private static final int DEFAULT_CAPACITY = 10;
+    private int newCapacity;
+    private T[] list;
+    private int size;
+
+    public ArrayList() {
+        newCapacity = DEFAULT_CAPACITY;
+        list = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity(size + 1);
+        list[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        ensureCapacity(size + 1);
+        if (index < 0 || index > size - 1) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        for (int i = size; i > index; i--) {
+            list[i] = list[i - 1];
+        }
+        list[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        ensureCapacity(size + 1);
+        for (int i = 0; i < list.size(); i++) {
+            this.list[size++] = list.get(i);
```

Comment:
```suggestion
            add(list.get(i));
```
in this case you don't need to to ensureCapacity on 44 line